File: CS/DXWebApplication1/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using DXWebApplication1.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace DXWebApplication1.Controllers
{
    public class HomeController : Controller
    {
        // GET: Home
        public ActionResult Index() {
            return View(PersonsList.GetPersons());
        }

        public ActionResult GridViewPartial() {
            return PartialView(PersonsList.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult BatchEditingUpdateModelPerson(MVCxGridViewBatchUpdateValues<Person, int> batchValues) {
            foreach(var person in batchValues.Update) {
                if(batchValues.IsValid(person))
                    PersonsList.UpdatePerson(person);
                else
                    batchValues.SetErrorText(person, "Correct validation errors");
            }

            foreach(var person in batchValues.Insert) {
                if(batchValues.IsValid(person))
                    PersonsList.AddPerson(person);
                else
                    batchValues.SetErrorText(person, "Correct validation errors");
            }

            foreach (var personID in batchValues.DeleteKeys) {
                PersonsList.DeletePerson(personID);
            }
            return PartialView("GridViewPartial", PersonsList.GetPersons());
        }

    }
}

File: VB/DXWebApplication1/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Web.Mvc
Imports DXWebApplication1.Models
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace DXWebApplication1.Controllers
	Public Class HomeController
		Inherits Controller
		' GET: Home
		Public Function Index() As ActionResult
			Return View(PersonsList.GetPersons())
		End Function

		Public Function GridViewPartial() As ActionResult
			Return PartialView(PersonsList.GetPersons())
		End Function

		<HttpPost, ValidateInput(False)> _
		Public Function BatchEditingUpdateModelPerson(ByVal batchValues As MVCxGridViewBatchUpdateValues(Of Person, Integer)) As ActionResult
			For Each person In batchValues.Update
				If batchValues.IsValid(person) Then
					PersonsList.UpdatePerson(person)
				Else
					batchValues.SetErrorText(person, "Correct validation errors")
				End If
			Next person

			For Each person In batchValues.Insert
				If batchValues.IsValid(person) Then
					PersonsList.AddPerson(person)
				Else
					batchValues.SetErrorText(person, "Correct validation errors")
				End If
			Next person

			For Each personID In batchValues.DeleteKeys
				PersonsList.DeletePerson(personID)
			Next personID
			Return PartialView("GridViewPartial", PersonsList.GetPersons())
		End Function

	End Class
End Namespace